---
layout: post
title:  "Humanization of Yeast Genes (Part 1)"
date:   2015-05-27 
tag:
- Yeast
- Biology
category: blog
author: jamie
---

<p>As I mentioned in my post before, yeast can be a really useful disease model for drug screens. Yeast screens are cheaper than screens using other organisms and you can go through a lot of compounds in very little time. Given that yeast and humans are separated by billions of years of evolution, a relevant question when considering yeast as a disease model is whether the results are translatable, and whether genes retain function after so many years of evolution. </p>

<p>A paper published on the May 22 issue of Science showed that yes, many genes do retain function. </p>

<p>The paper describes how a team of researchers from the Marcotte lab replaced 414 essential yeast genes with their human counterparts. To do this they cloned the human orthologs into a yeast plasmid and the genes were placed under control  of an inducible (Gal) or constitutive (GPD) promoter. Additionally, the yeast strains were engineered so that the expression of the yeast genes could be downregulated (with a tetracycline repressive promoter) or inactivated (via a temperature sensitive allele). They confirmed that the loss of the relevant gene caused a growth defect, and they found that replacing the yeast gene with the human version of the gene rescued growth in almost half of the genes (47%). </p>

<p>After seeing that only a fraction of the genes could be replaced it was relevant to see what characteristics of the genes determine their ability to be replaced by the homologs. </p>

<p>An interesting result was that sequence similarity doesn’t fully predict replaceability. They found sequence similarity was relevant with genes that had &gt; 50% sequence similarity or &lt; 20% (sequence similarity ranges from 9 to 92%, with an average of 32%) but in most pairs that fell into the 20-50% range the replaceability was poorly predicted. Instead replaceability seemed to be best predicted in terms of gene modules: genes in the same pathway or complex seemed to have a similar replaceability. Replaceable genes also had a tendency to be shorter and more highly expressed. Below you can see a figure showing what percentage of genes was replaceable based on gene function.</p>

<p><a href="https://svbtleusercontent.com/jpxfiockhrc8a.png" rel="nofollow"><img src="https://svbtleusercontent.com/jpxfiockhrc8a_small.png" alt="Screen Shot 2015-05-27 at 3.59.39 PM.png"></a></p>
<h2 id="a-follow-up-project_2">
<a class="head_anchor" href="#a-follow-up-project_2" rel="nofollow"> </a>A follow up project</h2>
<p>As part of my internship at Perlstein Lab (which started yesterday!) I’ll try to determine which of the essential yeast genes that the paper found can be replaced by their human counterpart are associated with a monogenic disease. Then I’ll identify the mutations in these genes that are known to cause disease. Then based on sequence alignments between yeast, worm, fly, zebrafish, mouse and human, I’ll figure out if disease-causing mutations affect conserved or variable amino acid positions. </p>

<p>Subscribe to my blog for updates on this project and other projects I’ll be working on throughout my internship! </p>
<h2 id="references_2">
<a class="head_anchor" href="#references_2" rel="nofollow"> </a>References</h2>
<p>Kachroo, AH et al. Systematic humanization of yeast genes reveals conserved functions and genetic modularity. Science <strong>348</strong>(6237) 921-925 (2015) </p>
